code/game/objects/items/devices/laserpointer.dm

```diff
@@ -44,128 +64,226 @@
+				crystal_stack.use_tool(src, user, amount = 1) //use only one if we were installing from a stack of crystals
+			return
+		//the single crystal that we actually install
+		var/obj/item/stack/ore/bluespace_crystal/single_crystal = crystal_stack.split_stack(null, 1)
+		if(isnull(single_crystal))
+			return
+		if(!user.transferItemToLoc(single_crystal, src))
+			return
+		crystal_lens = single_crystal
+		playsound(src, 'sound/items/screwdriver2.ogg', 30)
+		balloon_alert(user, "installed \the [crystal_lens.name]")
+		to_chat(user, span_notice("You install a [crystal_lens.name] in [src]. \
+			It can now be used to shine through obstacles at the cost of double the energy drain."))
+		return TRUE
+
+	return ..()
 
 /obj/item/laser_pointer/examine(mob/user)
 	. = ..()
 	if(in_range(user, src) || isobserver(user))
-		if(!diode)
+		if(isnull(diode))
 			. += span_notice("The diode is missing.")
 		else
 			. += span_notice("A class <b>[diode.rating]</b> laser diode is installed. It is <i>screwed</i> in place.")
+		. += span_notice("A small display reads out that[recharge_locked ? " it is currently recharging to full, and" : ""] there is <b>[energy * 10]%</b> total charge remaining.")
+		if(crystal_lens)
+			. += span_notice("There is a <b>[crystal_lens.name]</b> fit neatly before the focus lens. It can be <i>plucked out</i> with some <i>wirecutters</i>.")
+		else if(diode) //hint at the ability to modify the pointer with a crystal only if we have a diode
+			. += span_notice("<i>You could examine it more thoroughly...</i>")
+
+/obj/item/laser_pointer/examine_more(mob/user)
+	. = ..()
+	if(!isnull(crystal_lens) || isnull(diode))
+		return
+	switch(diode.rating)
+		if(1)
+			. += "<i>\The [diode.name] is fit neatly into the casing.</i>"
+		if(2)
+			. += "<i>\The [diode.name] is secured in place, with a little bit of room left between it and the focus lens.</i>"
+		if(3 to 4)
+			. += "<i>\The [diode.name]'s size is much smaller compared to the previous generation lasers, \
+			and the wide margin between it and the focus lens could probably house <b>a crystal</b> of some sort.</i>"
 
 /obj/item/laser_pointer/afterattack(atom/target, mob/living/user, flag, params)
 	. = ..()
 	. |= AFTERATTACK_PROCESSED_ITEM
 	laser_act(target, user, params)
 
+///Handles shining the clicked atom,
 /obj/item/laser_pointer/proc/laser_act(atom/target, mob/living/user, params)
-	if( !(user in (viewers(7,target))) )
-		return
-	if (!diode)
+	if(isnull(diode))
 		to_chat(user, span_notice("You point [src] at [target], but nothing happens!"))
 		return
-	if (!ISADVANCEDTOOLUSER(user))
+	if(!ISADVANCEDTOOLUSER(user))
 		to_chat(user, span_warning("You don't have the dexterity to do this!"))
 		return
 	if(HAS_TRAIT(user, TRAIT_CHUNKYFINGERS))
 		to_chat(user, span_warning("Your fingers can't press the button!"))
 		return
+
+	if(!(user in (view(7, target)))) //check if we are visible from the target's PoV
```

Comment:
Beside the fact this checks a hardcoded value of 7 rather than the client view, doesn't this mean a mob with thermal or x-ray vision could be able to shine a laser light everywhere on the station with the help or a (advanced) camera console?

Reply:
nope, they point at a screen object that tgui shows in its window